frontend/src/components/@common/Button/Button.tsx

```diff
@@ -0,0 +1,167 @@
+// import { HomeIcon } from 'assets/icons';
+import {
+  CSSProperties,
+  ComponentPropsWithRef,
+  ForwardedRef,
+  ReactElement,
+  forwardRef,
+} from 'react';
+
+import { css, styled } from 'styled-components';
+import { RuleSet } from 'styled-components/dist/types';
+
+import type { Color } from 'styles/theme';
+
+// Type
+export const Variant = ['primary', 'text'] as const;
+export type Variant = (typeof Variant)[number];
+
+export const Size = ['small', 'medium', 'large'] as const;
+export type Size = (typeof Size)[number];
+
+export const Align = ['left', 'center', 'right'] as const;
+export type Align = (typeof Align)[number];
+
+export type Props = {
+  variant?: Variant;
+  size?: Size;
+  backgroundColor?: Color;
+  fontColor?: Color;
+  block?: boolean;
+  align?: Align;
+  icon?: ReactElement;
+} & ComponentPropsWithRef<'button'>;
+
+// Component
+const Button = (
+  {
+    children,
+    variant = 'primary',
+    size = 'medium',
+    block = false,
+    align = 'center',
+    icon,
+    ...rest
+  }: Props,
+  ref: ForwardedRef<HTMLButtonElement>,
+) => {
+  return (
+    <S.Button ref={ref} variant={variant} size={size} block={block} align={align} {...rest}>
+      <S.IconTextContainer>
+        {Boolean(icon) && <S.IconWrapper size={size}>{icon}</S.IconWrapper>}
```

Comment:
이거 어차피 icon 없으면 undefined인데 이렇게 써도 되지 않을까요?
```suggestion
        {icon && <S.IconWrapper size={size}>{icon}</S.IconWrapper>}
```

Reply:
`icon`이 `undefined`인 경우 `undefined`는 falsy한 값이라 `false`인 `Boolean(icon)`과 동일하게 동작하겠지만,
저는 정확한 boolean 값을 이용하는게 좋다고 생각합니다. 
예상치 못한 동작을 막을 수 있기도 하고 더 직관적이라고 생각하는데, 어떻게 생각하시나요?

Reply:
흠.. 만약 count의 타입이 `undefined | number`라고 치면 조건문에서 Boolean 값을 이용하게 되면 0도 `false`로 동작하게 되는데 오히려 `count !== undefined` 이런식으로 아예 직관적으로 쓰는 건 어떤가요?

여기서는 icon은 무조건 `undefined | ReactElement`니까 `icon &&` 으로 쓰는것도 나쁘지 않다고 생각하긴 했는데 편한대로 써도 될 거 같아요!